src/sync/deployment.test.ts

```diff
@@ -30,6 +37,187 @@ describe("sync/deployments", () => {
+				const clone = JSON.parse(JSON.stringify(deploymentNodesFixture.data.repository.deployments.edges[0]));
+				clone._seq = idx + 1;
+				clone.cursor = `cursor:${idx + 1}`;
+				clone.node.createdAt = `2023-01-0${idx + 1}T10:00:00Z`;
+				clone.node.databaseId = `dbid-${idx + 1}`;
+				clone.node.commitOid = `SHA${idx + 1}`;
+				clone.node.latestStatus.createdAt = clone.node.createdAt;
+				return clone;
+			});
+		};
+		const getDeploymentsPageResponse = (items) => ({ "data": { "repository": { "deployments": { "edges": items } } } });
+		const nockFetchingDeploymentgPagesGraphQL = (cursor, deployments) => {
+			githubNock.post("/graphql", { query: getDeploymentsQuery, variables: { owner: repoSyncState.repoOwner, repo: repoSyncState.repoName, per_page: PAGE_SIZE__TWO_ITEMS, cursor } })
+				.query(true).reply(200, getDeploymentsPageResponse(deployments));
+		};
+		const nockDeploymentListingApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				githubNock.get(`/repos/test-repo-owner/test-repo-name/deployments?environment=prod&per_page=10`)
+					.reply(200, deployments.map((item, idx) => ({
+						"id": item.node.databaseId,
+						"sha": item.node.commitOid,
+						"ref": "random",
+						"task": `task for deployment ${idx + 1}`,
+						"payload": {},
+						"original_environment": item.node.environment,
+						"environment": item.node.environment,
+						"description": `description for deployment ${idx + 1}`,
+						"creator": { "login": "test-repo-owner", "id": 1, "type": "User" },
+						"created_at": item.node.createdAt,
+						"updated_at": item.node.createdAt,
+						"statuses_url": "random",
+						"repository_url": "random",
+						"transient_environment": false,
+						"production_environment": true
+					})));
+				deployments.forEach((item, idx) => {
+					githubNock.get(`/repos/test-repo-owner/test-repo-name/deployments/${item.node.databaseId}/statuses?per_page=100`)
+						.reply(200, { "id": idx, "state": "success", "description": "random", "environment": item.node.environment });
+				});
+			});
+			deployments.forEach((item, idx) => {
+				githubNock.get(`/repos/test-repo-owner/test-repo-name/commits/${item.node.commitOid}`)
+					.reply(200, {
+						commit: {
+							author: { name: "random", email: "random", date: new Date() },
+							message: `commit message [JIRA-${idx + 1}]`
+						}, html_url: "random"
+					});
+			});
+		};
+		const nockDeploymentCommitGetApi = (deployments, repeatTimes) => {
+			"*".repeat(repeatTimes).split("").forEach(() => {
+				deployments.forEach(item => {
+					githubNock.get(`/repos/test-repo-owner/test-repo-name/commits/${item.node.commitOid}`)
+						.reply(200, {
+							commit: {
+								author: { name: "random", email: "random", date: new Date() },
+								message: `commit message [JIRA-${item._seq}]`
+							}, html_url: "random"
+						});
+				});
+			});
+		};
+		const expectDeploymentEntryInDB = async (deployments) => {
```

Comment:
nit: can we add a new line between each variable declaration and each test? There's a lot of code here and the readability really drops when everything is clumped together without any spaces in between.

Reply:
Have add spaces in between, shuffle them a round, put some functions inline.
Hopefully it can be easier to read.